### DISCUSSION TITLE
`hg --files-with-matches` only makes the first file path clickable

### DISCUSSION THREAD

**Discussion by SirWrexes** [kovidgoyal/kitty#5742] [Q&A] [answered]
Sometimes when I look for the occurrences of a bit of text in my files, it's not to find the _lines in each file_ but rather the _list of files that include the text_.
Currently, using `hg` (alias for the `hyperlinked_grep` kitten) defined as per the documentation, passing the flag `-l`  or `--files-with-matches` doesn't work properly.
Only the first path is clickable, the rest is not and doesn't even behave as links -- they don't get underlined when I hover over them. 
I'm thinking this might be a bug.

**Comment by page-down** [comment]
I looked at every single option of ripgrep and here are all the ones that need special handling (version 13.0.0).

```text
--context-separator <SEPARATOR>
-c, --count
--count-matches
--field-context-separator <SEPARATOR>
--field-match-separator <SEPARATOR>
--files
-l, --files-with-matches
--files-without-match
-h, --help
--json
-I, --no-filename
--no-heading
-0, --null
--null-data
--path-separator <SEPARATOR>
--stats
--type-list
-V, --version
--vimgrep
```

For example skip `--version`, `--json`. And `--help` can be displayed using pager (keep the last screen content on the screen after exit).
(Or maybe don't change the behavior of `--help`, since the kitten is only focused on adding hyperlinks to the content.)
I haven't had time to deal with each one yet.

**Comment by kovidgoyal** [comment, marked as answer]
As is noted in the docs for the kitten. If you use options that change
the output format of rg it wont work, since the kitten functions by parsing
that output.

The proper solution for this is to get rg to output hyperlinks itself.
I'm not interested in trying to track all its output modes and implement
separate code for each of them. If you are, patches are most welcome.
Though at that point it would be easier to just fork rg or write a
replacement for it from scratch.

**Reply by SirWrexes** [reply]
Guess I didn't read the docs with enough attention.
I'm sorry for bothering you. Thanks for the fast reply ! :sweat_smile:
